BE/src/script_gen/nodes/news_research.py

```
@@ -194,27 +201,76 @@ def _filter_relevant_articles(articles: List[Dict], topic: str, search_keywords:
+                for art in b_articles[:b_count]:
+                    result.append(art)
+                
+                logger.info(f"A기사 {min(len(a_articles), a_count)}개, B기사 {min(len(b_articles), b_count)}개 추가")
+            
+            if not result:
                 logger.warning("필터 결과 0개 → 원본 유지")
                 return articles
             
-            return filtered
+            logger.info(f"최종 선별: {len(result)}개 (A+B:{len([i for i in ab_indices if 0 <= i < len(articles)])}, A:{len(a_only_indices)}, B:{len(b_only_indices)})")
+            return result
```

Ruff (0.14.14)

[warning] Consider moving this statement to an `else` block

(TRY300)